core/src/main/java/com/linecorp/armeria/server/healthcheck/DefaultCpuHealthChecker.java

```diff
@@ -0,0 +1,122 @@
+import java.lang.management.OperatingSystemMXBean;
+import java.lang.reflect.Method;
+import java.util.List;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.common.annotation.Nullable;
+
+/**
+ * The Default cpu health checker.<br>
+ * Here's an example of how to use it.<br>
+ * final DefaultCpuHealthChecker cpuHealthChecker = HealthChecker.of(10, 10);<br>
+ * final boolean healthy = cpuHealthChecker.isHealthy();
+ */
+// Forked from <a href="https://github.com/micrometer-metrics/micrometer/blob/8339d57bef8689beb8d7a18b429a166f6595f2af/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/system/ProcessorMetrics.java">ProcessorMetrics.java</a> in the micrometer core.
+final class DefaultCpuHealthChecker implements HealthChecker {
+
+    private static final List<String> OPERATING_SYSTEM_BEAN_CLASS_NAMES = ImmutableList.of(
+            "com.ibm.lang.management.OperatingSystemMXBean", // J9
+            "com.sun.management.OperatingSystemMXBean" // HotSpot
+    );
+
+    private final OperatingSystemMXBean operatingSystemBean;
+
+    private final Class<?> operatingSystemBeanClass;
+
+    @Nullable
+    @VisibleForTesting
+    final Method systemCpuUsage;
+
+    private final double targetCpuUsage;
+
+    @Nullable
+    @VisibleForTesting
+    final Method processCpuUsage;
+
+    private final double targetProcessCpuLoad;
+
+    /**
+     * Instantiates a new Default cpu health checker.
+     *
+     * @param targetCpuUsage the target cpu usage
+     * @param targetProcessCpuLoad the target process cpu usage
+     */
+    private DefaultCpuHealthChecker(int targetCpuUsage, int targetProcessCpuLoad) {
+        this.targetCpuUsage = targetCpuUsage;
+        this.targetProcessCpuLoad = targetProcessCpuLoad;
+        this.operatingSystemBean = ManagementFactory.getOperatingSystemMXBean();
+        this.operatingSystemBeanClass = requireNonNull(getFirstClassFound(OPERATING_SYSTEM_BEAN_CLASS_NAMES));
+        this.systemCpuUsage = detectMethod("getSystemCpuLoad");
+        this.processCpuUsage = detectMethod("getProcessCpuLoad");
+    }
+
+    public static DefaultCpuHealthChecker of(int targetCpuUsage, int targetProcessCpuLoad) {
+        return new DefaultCpuHealthChecker(targetCpuUsage, targetProcessCpuLoad);
+    }
+
+    /**
+     * Returns true if and only if System CPU Usage and Processes cpu usage is below the target usage.
+     */
+    @Override
+    public boolean isHealthy() {
+        final double currentSystemCpuUsage = invoke(systemCpuUsage);
+        final double currentProcessCpuUsage = invoke(processCpuUsage);
+        return currentSystemCpuUsage <= targetCpuUsage && currentProcessCpuUsage <= targetProcessCpuLoad;
+    }
+
+    private double invoke(final Method method) {
+        try {
+            final MethodHandles.Lookup lookup = MethodHandles.lookup();
+            final MethodHandle mh = lookup.unreflect(method);
```

Comment:
Could we keep the `MethodHandle` in a static final field, so we don't have to look up on every invocation? Otherwise, we may have a leak of `MethodHandle`s. (There's a known JDK issue related with this.)

Reply:
Modified to statically generate or recycle using the cache.